test/shared/src/main/scala/zio/test/TestConfig.scala

```diff
@@ -48,6 +49,14 @@ trait TestConfig extends Serializable {
    * The maximum number of shrinkings to minimize large failures
    */
   def shrinks: Int
+
+  /**
+   * Aspect that should be applied to each check sample.
+   *
+   * NOTE: default implementation for backward compatibility. Remove in next
+   * major version.
+   */
+  def checkSampleAspect: TestAspect.CheckSampleAspect = ZIOAspect.identity
```

Comment:
This guy is a bit unfortunate as it's possible to forget to forward the aspect when creating proxies like in TestAspect.scala, but it's the most backward-compatible approach I could come up with.
Suggestions welcome.

Reply:
Had this issue myself a few times.

Scala needs an annotation on methods that should always be overwritten by concrete implementations 🥲